pydantic/_internal/_generate_schema.py

```diff
@@ -1497,7 +1498,6 @@ def _get_prepare_pydantic_annotations_for_known_type(
     ) -> tuple[Any, list[Any]] | None:
         from ._std_types_schema import PREPARE_METHODS
 
-        # This check for hashability is only necessary for python 3.7
```

Comment:
Seem we need this check for other versions as well. by removing this check some tests fail. like https://github.com/pydantic/pydantic/blob/8989f96415b0c693585fa9363eaf520d5b9a30a2/tests/test_json_schema.py#L4891

Reply:
please update the comment.